Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -332,34 +334,54 @@ public void FailPlugin(Assembly assembly)
             ChangePluginState(plugin, PluginStatus.Malfunctioned);
         }
 
-        /// <summary>
-        /// Saves the manifest back to disk.
-        /// </summary>
-        /// <param name="manifest">The <see cref="PluginManifest"/> to save.</param>
-        /// <param name="path">The path where to save the manifest.</param>
-        /// <returns>True if successful.</returns>
+        /// <inheritdoc/>
         public bool SaveManifest(PluginManifest manifest, string path)
         {
-            if (manifest == null)
-            {
-                return false;
-            }
-
             try
             {
                 var data = JsonSerializer.Serialize(manifest, _jsonOptions);
                 File.WriteAllText(Path.Combine(path, "meta.json"), data);
                 return true;
             }
-#pragma warning disable CA1031 // Do not catch general exception types
-            catch (Exception e)
-#pragma warning restore CA1031 // Do not catch general exception types
+            catch (ArgumentException e)
             {
-                _logger.LogWarning(e, "Unable to save plugin manifest. {Path}", path);
+                _logger.LogWarning(e, "Unable to save plugin manifest due to invalid value. {Path}", path);
                 return false;
             }
         }
 
+        /// <inheritdoc/>
+        public bool GenerateManifest(PackageInfo packageInfo, Version version, string path)
+        {
+            var versionInfo = packageInfo.Versions.First(v => v.Version == version.ToString());
+            var url = packageInfo.ImageUrl ?? string.Empty;
+            var imageFilename = url.Substring(url.LastIndexOf('/') + 1, url.Length);
+
+            using (var client = new WebClient())
```

Comment:
We use `HttpClient`

Something like:
```c#
            await using var fileStream = File.OpenWrite(Path.Join(path, imageFilename));
            var downloadStream = await _httpClientFactory.CreateClient(NamedClient.Default)
                .GetStreamAsync(url)
                .ConfigureAwait(false);
            await downloadStream.CopyToAsync(fileStream).ConfigureAwait(false);
```

You can get `IHttpClientFactory` in the `PluginManager` constructor

Reply:
This requires async which the PluginManager doesn't use yet, do you want it added?

Reply:
If it's not too much work, the function should be `async` as well.
If too many other changes are required `.GetAwaiter().GetResult()` can be used with a `TODO` note

Reply:
This whole pull request is a TODO until we add these fields to the database :laughing: but I'll add the async now anyways. It's marked as a draft because I haven't tested it as thoroughly as possible yet, any chance you have any edge cases you want me to confirm?

Reply:
I would test valid config, config without image, and config where the image url isn’t an image (folder or other random file)